src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseServiceClient.java

```diff
@@ -56,16 +51,28 @@ public Mono<Hearing> getHearing(String hearingId) {
                 .retrieve()
                 .onStatus(HttpStatusCode::isError, (clientResponse) -> handleGetError(clientResponse, hearingId))
                 .bodyToMono(CCSExtendedHearing.class)
-                .map(response -> {
-                    log.debug("GET succeeded for the hearing at {}", path);
-                    return response.asDomain();
-                })
+                .map(CCSExtendedHearing::asDomain)
                 .onErrorResume(HearingNotFoundException.class, (e) -> {
                     // This is normal in the context of CCM, don't log
                     return Mono.empty();
                 });
     }
 
+    public Mono<Hearing> getHearing(String hearingId, String courtCaseId) {
+        final String path = String.format("/hearing/%s/case/%s", hearingId, courtCaseId);
+
+        // Get the existing case. Not a problem if it's not there. So return a Mono.empty() if it's not
```

Comment:
There are 3 comments in this flow saying more or less this - seems like a lot.

Reply:
Agreed, removed the comment.